.github/workflows/windows-builds.yml

```diff
@@ -0,0 +1,212 @@
+          echo "GIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "GIT_TAG=$(git describe --tags --abbrev=0)" >> $GITHUB_ENV
+      - name: Restore Qt from cache
+        id: cache-qt
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ runner.workspace }}/Qt32
+          key: ${{ runner.os }}-QtCache-32bit
+      - name: 'Install 32-bit Qt'
+        uses: jurplel/install-qt-action@v2
+        with:
+          cached: ${{ steps.cache-qt.outputs.cache-hit }}
+          version: '${{ env.QT_VERSION }}'
+          arch: 'win32_msvc2017'
+          dir: ${{ runner.workspace }}/Qt32
+      - name: 'Restore from cache and run vcpkg'
+        uses: lukka/run-vcpkg@v5
+        with:
+          vcpkgArguments: '@${{ github.workspace }}/vcpkg.txt'
+          vcpkgDirectory: '${{ github.workspace }}/vcpkg'
+          appendedCacheKey: ${{ hashFiles(env.vcpkgResponseFile) }}
+      - name: 'Configure Cockatrice 32-bit'
+        working-directory: ${{ github.workspace }}
+        run: |
+          New-Item build32 -type directory -force
+          cd .\build32
+          cmake -G "${{ env.CMAKE_GENERATOR }}" -A "Win32" -DCMAKE_TOOLCHAIN_FILE="${{ github.workspace }}\vcpkg\scripts\buildsystems\vcpkg.cmake" -DCMAKE_PREFIX_PATH="${{ runner.workspace }}\Qt32\Qt\5.12.9\msvc2017;${{ github.workspace }}\vcpkg\installed\x86-windows\bin" -DCMAKE_BUILD_TYPE="Release" -DWITH_SERVER=1 -DTEST=1 ..
+      - name: 'Build Cockatrice 32-bit'
+        working-directory: ${{ github.workspace }}
+        run: msbuild /m /p:Configuration=Release .\build32\Cockatrice.sln
+      - name: 'Build Cockatrice Installer Package 32-bit'
+        working-directory: ${{ github.workspace }}
+        run: |
+          cd build32
+          msbuild /m /p:Configuration=Release PACKAGE.vcxproj
+          cp *.exe ../Cockatrice-${{ env.GIT_TAG }}-${{ env.GIT_HASH }}-32bit-installer.exe
+      - name: 'Run Tests'
+        working-directory: ${{ github.workspace }}/build32
+        run: ctest -T Test -C Release
+      - name: 'Publish'
+        if: success()
+        uses: actions/upload-artifact@v2
+        with:
+          name: 'Cockatrice-${{ env.GIT_TAG }}-32bit'
+          path: './*.exe'
+  make-release:
+    name: 'Create and upload release'
+    runs-on: [ubuntu-latest]
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    needs: [win32,win64]
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - name: 'Fetch git tags and generate file name'
+        shell: bash
+        run: |
+          git fetch --prune --unshallow
+          echo "GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)" >> $GITHUB_ENV
+          echo "GIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "GIT_TAG=$(git describe --tags --abbrev=0)" >> $GITHUB_ENV
+      - name: 'Checking if beta'
+        if: contains(env.GIT_TAG, 'beta')
```

Comment:
can we not do an if/else here, instead of having two name operations?

Reply:
This is typically "safer" and won't really add any time to the overall workflow as one is always skipped.